dsc/examples/file_with_condition.dsc.bicep

```diff
@@ -0,0 +1,38 @@
+// This example demonstrates how to create a file using the Windows PowerShell DSC extension.
+// The file is created in the C:\DSC directory on the target machine.
+// You should at least have the Bicep CLI v0.34.34 installed to run this example with experimental feature desiredStateConfiguration turned on.
+// To run the second resource, you can add the --parameters '{"parameters":{"restartService":true}}' flag to the command line.
+
+targetScope = 'desiredStateConfiguration'
+
+@description('Set to true to ensure the service is running after the file creation.')
+param restartService bool = false
+
+resource powerShellAdapter 'Microsoft.Windows/WindowsPowerShell@2025-01-07' = {
```

Comment:
You should be able to use the File resource implicitly instead of explicitly using the adapter which I think will be more common usage